### PR TITLE
[Merged by Bors] - chore: forward-port leanprover-community/mathlib#18856

### PR DESCRIPTION
**Pull request by ericrbg**
Forward-ports https://github.com/leanprover-community/mathlib/pull/18856 and https://github.com/leanprover-community/mathlib/pull/18858

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

https://github.com/leanprover-community/mathlib/pull/18856 & https://github.com/leanprover-community/mathlib/pull/18858 for reviewer reference

### CODE REVIEW

File: Mathlib/Data/ZMod/Basic.lean [hunk 1]
```diff
@@ -4,13 +4,13 @@ Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Chris Hughes
 
 ! This file was ported from Lean 3 source module data.zmod.basic
-! leanprover-community/mathlib commit 47a1a73351de8dd6c8d3d32b569c8e434b03ca47
+! leanprover-community/mathlib commit 473ee9e095b89a24df125a4d363bd996f0989748
```
Comment (eric-wieser) [inline Mathlib/Data/ZMod/Basic.lean:7]: The dashboard isn't even aware of https://leanprover-community.github.io/mathlib-port-status/file/data/zmod/basic?range=47a1a73351de8dd6c8d3d32b569c8e434b03ca47..473ee9e095b89a24df125a4d363bd996f0989748 yet!

Comment (ericrbg) [reply Mathlib/Data/ZMod/Basic.lean:7]: I just did this manually using the SHAs, should I have just let mathport build & etc? Thought it wouldn't be that bad as a one-line change

Comment (eric-wieser) [reply Mathlib/Data/ZMod/Basic.lean:7]: No, this one was fine to do manually; the dashboard should pick up the mathlib3 commit info in the next 15 minutes. There's certainly no need to wait for mathport for this PR.

File: Mathlib/Data/ZMod/Basic.lean [hunk 2]
```diff
@@ -4,13 +4,13 @@ Released under Apache 2.0 license as described in the file LICENSE.
...
 ! Please do not edit these lines, except to modify the commit id
 ! if you have ported upstream changes.
 -/
 import Mathlib.Algebra.CharP.Basic
+import Mathlib.Data.Fintype.Units
```
Comment (eric-wieser) [inline Mathlib/Data/ZMod/Basic.lean:12]: Strictly speaking these imports are in a different order in mathlib3; can you use the same order here?

Or alternatively, make a mathlib3 PR that fixes the order...

Comment (ericrbg) [reply Mathlib/Data/ZMod/Basic.lean:12]: https://github.com/leanprover-community/mathlib/pull/18858

**Comment by semorrison** [comment]
@ericrbg, you've written "DO NOT MERGE YET", yet it is marked for review. I'll leave it as `awaiting-author` for now.

**Comment by ericrbg** [comment]
I was waiting for the SHA from the second alphabetising PR - all good! thanks scott :)

**Review by eric-wieser — APPROVED** [review]
bors merge

Thanks!

**Pull request closed without merging**